Model/EpsPaymentMethod.php

```diff
@@ -0,0 +1,151 @@
+            EpsTransaction::NAME,
+            $this->_oPayment->oxpayments__wdoxidee_maid->value,
+            $this->_oPayment->oxpayments__wdoxidee_secret->value
+        );
+
+        $oConfig->add($oPaymentMethodConfig);
+
+        return $oConfig;
+    }
+
+    /**
+     * Get the current transaction to be processed
+     *
+     * @return Transaction
+     *
+     * @since 1.2.0
+     */
+    public function getTransaction()
+    {
+        return new EpsTransaction();
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getConfigFields()
+    {
+        $aAdditionalFields = [
+            'descriptor' => [
+                'type'  => 'select',
+                'field' => 'oxpayments__wdoxidee_descriptor',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_descriptor'),
+                'description' => Helper::translate('wd_config_descriptor_desc'),
+            ],
+            'additionalInfo' => [
+                'type'  => 'select',
+                'field' => 'oxpayments__wdoxidee_additional_info',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_additional_info'),
+                'description' => Helper::translate('wd_config_additional_info_desc'),
+            ],
+            'deleteCanceledOrder' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_delete_canceled_order',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_delete_cancel_order'),
+                'description' => Helper::translate('wd_config_delete_cancel_order_desc'),
+            ],
+            'deleteFailedOrder' => [
+                'type' => 'select',
+                'field' => 'oxpayments__wdoxidee_delete_failed_order',
+                'options' => [
+                    '1' => Helper::translate('wd_yes'),
+                    '0' => Helper::translate('wd_no'),
+                ],
+                'title' => Helper::translate('wd_config_delete_failure_order'),
+                'description' => Helper::translate('wd_config_delete_failure_order_desc'),
+            ],
+        ];
+
+        return array_merge(parent::getConfigFields(), $aAdditionalFields);
+    }
+
+    /**
+     * @inheritdoc
+     *
+     * @return array
+     *
+     * @since 1.2.0
+     */
+    public function getPublicFieldNames()
+    {
+        return array_merge(parent::getPublicFieldNames(), [
+            'additionalInfo',
```

Comment:
please also add the descriptor or is there a special reason to not add it

Reply:
Done :+1: